### PR TITLE
issue #35 Option to self-generate a keystore when building or use a previously …

### PR DESCRIPTION
**Pull request by ddelblanco**
…created one

### CODE REVIEW

**Comment by ddelblanco** [comment]
To make this work in travis we need to delete the equella.keystore file or... set the build.in properties to the original password.

**Comment by ddelblanco** [comment]
Updated the README.md to explain about the self-generated keystore.

File: README.md [hunk 1]
```diff
@@ -19,17 +19,16 @@ The build artifacts will be produced in the `product` folder.
 
 **NOTE: KEYSTORE**
 A keystore with a certificate is required to sign some of the jars or the build will fail.
-In the build.properties.in file you can modify the parameters to configure your own keystore. the needed ones are:
+In the build.properties.in file you can modify the parameters to configure your own keystore:
```
Comment (doolse) [inline README.md]: build.properties.in shouldn't be modified, it's only for defaults - only build.properties should be modified after: 
`ant generate-build-properties`. However I see the problem with `ant release` is that it always runs the `ant generate-build-properties` - it should probably only do this if `build.properties` doesn't already exist.

Comment (ddelblanco) [inline README.md]: Yes, that was the problem, build.properties is always overwritten by the build.properties.in. I did the check about if the keystore file exists, so I can do the same here checking if the build properties exists.

**Comment by ddelblanco** [comment]
Seems that now it builds with the changes in the travis configuration and only overwrites build.properties if it doesn't exits.

File: build.xml [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+<project name="EQUELLA-Build" default="build" basedir="." xmlns:if="ant:if"
+    xmlns:unless="ant:unless">
 	<property name="mm" value="6.4" />
 
-	<target name="generate-build-properties">
-		<delete file="build.properties" />
+	<target name="build-properties-exists">
+		<condition property="buildPropExists">     
+	    		<and>
+	    			<available file="build.properties}"/>
```
Comment (doolse) [inline build.xml]: There's an extra trailing "}" here, so I don't think this is having any effect. I believe the <copy> task doesn't overwrite if the file exists, so I think just removing the <delete> should be enough

**Comment by ddelblanco** [comment]
Yes, seems that removing the delete command was enough to just overwrite if it doesn't exist. Updated

**Pull request merged**